src/query/api/v1/handler/prometheus/remote/write.go

```diff
@@ -517,6 +557,36 @@ func (h *PromWriteHandler) parseRequest(
 		}
 	}
 
+	if h.remoteWriteOpts.rejectOldSamples {
+		now := h.nowFn()
+		thresholdTime := now.Add(-h.remoteWriteOpts.rejectDuration)
+		// The two for loops do the trick of not allocating new memory and do in place drops
+		i := 0
+		for _, ts := range req.Timeseries {
+			j := 0
+			var t int64
+			for _, s := range ts.Samples {
+				t = s.Timestamp
+				if time.UnixMilli(s.Timestamp).After(thresholdTime) {
```

Comment:
do we confident that all timeseries that we received in coordinator are in order? this we cut timeseries from a threshold and all remaining samples are older ?

Reply:
I don't think this requires samples in order, for example we have a sequence of `[A B A B B B]` and A is kept and B is dropped, the result will be `[A A A B B B]` and we keep slice [:2] here.